test/exq_scheduler_test.exs

```diff
@@ -19,4 +28,103 @@ defmodule ExqSchedulerTest do
+  test "check continuity" do
+    redis = redis_pid("continuity")
+    storage_opts = Storage.build_opts(env([:redis, :name], redis))
+    :timer.sleep(4000) # 3+ Hour
+
+    jobs = get_jobs_from_storage(redis, Storage.queue_key("default", storage_opts))
+           |> Enum.filter(fn job -> job.class == "DummyWorker1" end)
+
+    assert_continuity(jobs, 3600)
+  end
+
+  @tag config: configure_env(env(), 100, 1000*1200, [schedule_cron_1m: %{
+                                                   "cron" => "*/10 * * * * *",
+                                                   "class" => "DummyWorker2",
+                                                   "include_metadata" => true
+                                                }])
+  test "check for missing jobs" do
+    redis = redis_pid("missing_jobs")
+    storage_opts = Storage.build_opts(env([:redis, :name], redis))
+    :timer.sleep(4000) # 3+ Hour
+
+    jobs = get_jobs_from_storage(redis, Storage.queue_key("default", storage_opts))
+           |> Enum.filter(fn job -> job.class == "DummyWorker2" end)
+    assert_continuity(jobs, 10*60)
+  end
+
+  @tag config: configure_env(env(), 1000, 10000, [schedule_cron_1m: %{
+                                                   "cron" => "* * * * * *",
+                                                   "class" => "QWorker",
+                                                   "queue" => "SuperQ"
+                                                }])
+  test "Check schedules are getting added to correct queues" do
+    redis = redis_pid("queue")
+    storage_opts = Storage.build_opts(env([:redis, :name], redis))
+    :timer.sleep(3000) # 3+ Hour
+
+    jobs = get_jobs_from_storage(redis, Storage.queue_key("SuperQ", storage_opts))
+           |> Enum.filter(fn job -> job.class == "QWorker" end)
+    assert length(jobs) >= 1
+  end
+
+  @tag config: configure_env(env(), 10, 10000000, [schedule_cron_1h: %{
+                                                    "cron" => "0 * * * * *",
+                                                    "class" => "TimeWorker",
+                                                    "queue" => "TimeQ",
+                                                    "include_metadata" => true
+                                                 }])
+  test "scheduler should not consider dates before its started" do
+    redis = redis_pid("old_dates")
+    storage_opts = Storage.build_opts(env([:redis, :name], redis))
+    start_time = Timex.to_unix(Time.now())
+    :timer.sleep(100) # 3+ Hour
+
+    jobs =
+      get_jobs_from_storage(redis, Storage.queue_key("TimeQ", storage_opts))
+      |> Enum.filter(fn job -> job.class == "TimeWorker" end)
+
+    is_jobs_scheduled_before_start =
+      jobs
+      |> Enum.all?(
+         fn job ->
+           st = List.first(job.args)["scheduled_at"]
+           convert_from_iso_to_unixtime(st) > start_time
+         end)
+    assert(is_jobs_scheduled_before_start == true, inspect(jobs))
```

Comment:
== true is redundant

Reply:
You mean we can assert inside the 'Enum.all' itself?

Reply:
assert(x) is the same as assert(x == true)